crates/nu-command/src/filesystem/cd.rs

```diff
@@ -169,11 +171,17 @@ impl Command for Cd {
         if let Some(oldpwd) = stack.get_env_var(engine_state, "PWD") {
             stack.add_env_var("OLDPWD".into(), oldpwd)
         }
-
         //FIXME: this only changes the current scope, but instead this environment variable
         //should probably be a block that loads the information from the state in the overlay
 
-        stack.add_env_var("PWD".into(), path_value);
+        // get a fake path ,is can get it , or not permission_denied , then can cd int
+        // the path
+        if permission_denied(PathBuf::from(format!("{}/enter", checkpath))) {
+            return Err(ShellError::IOError("Cannot into ".to_string()));
```

Comment:
Maybe change this text from "Cannot into " to something more like "Permission denied changing to {} directory"